src/rpc/rpc_reward_api.py

```diff
@@ -83,10 +85,28 @@ def get_rewards_for_cycle_map(self, cycle, expected_rewards=False):
 
                 if current_level - level_of_last_block_in_unfreeze_cycle >= 0:
                     unfrozen_fees, unfrozen_rewards = self.__get_unfrozen_rewards(level_of_last_block_in_unfreeze_cycle, cycle)
-                    reward_data["total_rewards"] = unfrozen_fees + unfrozen_rewards
+                    total_actual_rewards = unfrozen_fees + unfrozen_rewards
                 else:
                     frozen_fees, frozen_rewards = self.__get_frozen_rewards(cycle, current_level)
-                    reward_data["total_rewards"] = frozen_fees + frozen_rewards
+                    total_actual_rewards = frozen_fees + frozen_rewards
+                if rewards_type.isActual():
+                    reward_data["total_rewards"] = total_actual_rewards
+                elif rewards_type.isIdeal():
+                    missed_baking_income = 0
+                    for r in self.__get_baking_rights(cycle, level_of_first_block_in_preserved_cycles):
+                        if r["priority"] == 0:
+                            if self.__get_block_author(r["level"]) != self.baking_address:
+                                logger.warning("Found missed baking slot {}, adding {} mutez reward anyway.".format(r, self.block_reward))
+                                missed_baking_income += self.block_reward
+                    missed_endorsing_income = 0
+                    for r in self.__get_endorsement_rights(cycle, level_of_first_block_in_preserved_cycles):
+                        authored_endorsement_slots = self.__get_authored_endorsement_slots_by_level(r["level"] + 1)
+                        if authored_endorsement_slots != r["slots"]:
+                            mutez_to_add = self.endorsement_reward * len(r["slots"])
+                            logger.warning("Found {} missed endorsement(s) at level {}, adding {} mutez reward anyway.".format(len(r["slots"]), r["level"], mutez_to_add))
+                            missed_endorsing_income += mutez_to_add
+                    logger.warning("total rewards %s" % (total_actual_rewards + missed_baking_income + missed_endorsing_income))
+                    reward_data["total_rewards"] = total_actual_rewards + missed_baking_income + missed_endorsing_income
```

Comment:
@dansan566 this must be validated by using the new reward_type ideal and comparing tzkt output with prpc output from a cycle with some lost blocks and endorements

Actually by comparing PRPC and tzkt for all 3 reward types after tzkt and tzstats have been confirmed of working the same way.